arroyo/types.py

```diff
@@ -85,6 +79,76 @@ def __setstate__(self, state: dict[str, int | datetime]) -> None:
             object.__setattr__(self, slot, value)
 
 
+class BasePayload(Generic[TPayload]):
+    @property
+    def payload(self) -> TPayload:
+        pass
+
+    @property
+    def committable(self) -> Mapping[Partition, Position]:
+        pass
+
+
+@dataclass(unsafe_hash=True)
+class Payload(BasePayload[TPayload]):
```

Comment:
These two classes will create a lot of instantiations on high volume consumers. 
I think profiling or a load test is mandatory or we could be cutting down the throughput of our consumers by an undetermined amount. 

I would suggest you run the indexer load test referencing this specific arroyo commit in the dependency file so you can test the change without having to release arroyo. the pypi requirements file can reference a github commit instead of a release.

Reply:
I ran a few tests locally. It is about 0.1 seconds slower per 100,000 messages on my machine. I think it's acceptable. Since we are only talking about the cost of an extra Python class, I am going to skip the full indexer load test.

Reply:
Please still run the load test once you have the new version of the api released. It only takes 20 minutes to run it.